activation/atxdb.go

```diff
@@ -578,11 +582,18 @@ func (db *ActivationDb) addAtxToNodeId(nodeId types.NodeId, atx *types.Activatio
 
 // GetNodeLastAtxId returns the last atx id that was received for node nodeId
 func (db *ActivationDb) GetNodeLastAtxId(nodeId types.NodeId) (types.AtxId, error) {
-	db.log.Debug("fetching atxIDs for node %v", nodeId.ShortString())
+	nodeAtxsIterator := db.atxs.Find(getNodeAtxPrefix(nodeId))
```

Comment:
you are assuming that the `nodeAtxsIterator.Last()` is in fact the last atx created / received, is this always true?

Reply:
Yes. The syntactic validation ensures that the previous ATX is at least an epoch before the current one and the contextual validation ensures that the previous ATX is the last one based on this method.

Now that you've asked, I think that it may have been a good idea to explain this in a code comment, so I'll add it now.